pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntime.java

```diff
@@ -585,6 +585,7 @@ private V1Container getFunctionContainer(List<String> instanceCommand, Function.
         requests.put("memory", Quantity.fromString(Long.toString(resource != null && resource.getRam() != 0 ? resource.getRam() : 1073741824)));
         requests.put("cpu", Quantity.fromString(Double.toString(resource != null && resource.getCpu() != 0 ? resource.getCpu() : 1)));
         resourceRequirements.setRequests(requests);
```

Comment:
Actually, if we are requesting to get at least the max resources, there can be overhead. Eg: each functions instance has 1 core max limit, therefore we're also requesting to guarantee at-min to have 1 core reserved.

We could just set the limit and not the request, or have a request much lower than the actual limit, and leave K8S to move pods around when several of them are using up to the max the resources.

Reply:
Not sure I concur with this. This actually can lead to very sticky situations. For instance imagine someone deploying a bunch of functions that all go into running mode because the minimum criteria is met but when the load starts up, some of them will be booted out and may never get scheduled if there are no resources left(this actually happened at several production deployments that I've worked with). Thus I believe that request/max should be the same.